### PR TITLE
Check for slot existence

### PR DESCRIPTION
**Pull request by ejoerns**
This adds the SlotState ST_NOTFOUND that will be set when the
destination slot path did not exist when calling determine_slot_states().

When trying to install into a slot with state ST_NOTFUND, the default
handler will report an error.

### CODE REVIEW

File: include/config_file.h [hunk 1]
```diff
@@ -28,6 +28,7 @@ typedef enum {
 	ST_ACTIVE = 1,
 	ST_INACTIVE = 2,
 	ST_BOOTED = 4 | ST_ACTIVE,
+	ST_NOTFOUND = 8 | ST_INACTIVE,
```
Comment (jluebbe) [inline include/config_file.h]: ST_NOTFOUND should not imply ST_INACTIVE

File: src/install.c [hunk 2]
```diff
@@ -191,6 +191,11 @@ gboolean determine_slot_states(void) {
 			if (s->state == ST_UNKNOWN)
 				s->state |= ST_INACTIVE;
 		}
+
+		if (!g_file_test(s->device, G_FILE_TEST_EXISTS)) {
```
Comment (jluebbe) [inline src/install.c]: Perform this check first, before checking if this slot has a parent.

File: src/install.c [hunk 3]
```diff
@@ -250,7 +255,7 @@ GHashTable* determine_target_install_group(RaucManifest *manifest) {
 	for (GList *l = slotmembers; l != NULL; l = l->next) {
 		RaucSlot *s = (RaucSlot*) g_hash_table_lookup(r_context()->config->slots, l->data);
 
-		if (s->state == ST_INACTIVE) {
+		if (s->state & ST_INACTIVE) {
```
Comment (jluebbe) [inline src/install.c]: We can keep this as == because we don't overload ST_INACTIVE anymore.

File: src/install.c [hunk 4]
```diff
@@ -613,6 +618,17 @@ static gboolean launch_and_wait_default_handler(RaucInstallArgs *args, gchar* cw
 	GHashTableIter iter;
 	gpointer class, member;
 
+	g_hash_table_iter_init(&iter, target_group);
```
Comment (jluebbe) [inline src/install.c]: Drop this. We won't have ST_NOTFOUND slots in the target_group.

File: src/main.c [hunk 5]
```diff
@@ -354,7 +357,7 @@ static gboolean status_start(int argc, char **argv)
 	}
 
 	if (argc < 3) {
-		r_exit_status = 1;
+		r_exit_status = 0;
```
Comment (jluebbe) [inline src/main.c]: Keep this as error code 1.

File: test/rauc.t [hunk 6]
```diff
@@ -69,12 +69,24 @@ test_expect_success "rauc bundle" "
 "
 
 
+TMPDIR=$(mktemp -d)
```
Comment (jluebbe) [inline test/rauc.t]: Check if this change is still needed.

File: src/install.c [hunk 7]
```diff
@@ -198,6 +198,12 @@ gboolean determine_slot_states(GError **error) {
 	for (GList *l = slotlist; l != NULL; l = l->next) {
 		RaucSlot *s = (RaucSlot*) g_hash_table_lookup(r_context()->config->slots, l->data);
 
+		if (!g_file_test(s->device, G_FILE_TEST_EXISTS)) {
+			g_message("Device for slot %s not found (%s)", s->name, s->device);
```
Comment (jluebbe) [inline src/install.c]: This should be a warning.

**Comment by ejoerns** [comment]
This one is pretty outdated and it is unsure if we still need this.

Anyway, if required, it might be worth simply re-implementing this on top of current code base.

Thus we close this for now.

**Pull request closed without merging**